src/features/projectsV2/components/ProjectSnippet.tsx

```diff
@@ -0,0 +1,153 @@
+}
+
+export interface ImageSectionProps extends CommonProps {
+  projectName: string;
+  image: string;
+  ecosystem: EcosystemTypes | null;
+  showPopup: boolean;
+  projectReviews: Review[] | undefined;
+  classification: TreeProjectClassification;
+}
+
+export interface ProjectInfoProps extends CommonProps {
+  unitType: 'm2' | 'tree';
+  countPlanted: number;
+  unitCost: number;
+  country: CountryCode;
+  currency: CurrencyCode;
+  slug: string;
+  unitsContributed?: number;
+}
+
+export default function ProjectSnippet({
+  project,
+  showPopup,
+}: Props): ReactElement {
+  const router = useRouter();
+  const tCommon = useTranslations('Common');
+  const { embed } = useContext(ParamsContext);
+
+  const ecosystem =
+    project._scope === 'map' ? project.ecosystem : project.metadata.ecosystem;
+  const isTopProject = project.purpose === 'trees' && project.isTopProject;
+  const isApproved = project.purpose === 'trees' && project.isApproved;
+
+  const progressPercentage = useMemo(() => {
+    if (project.purpose === 'trees' && project.countTarget) {
+      return Math.min((project.countPlanted / project.countTarget) * 100, 100);
+    }
+    return 0;
+  }, [
+    project.purpose === 'trees' && project.countPlanted,
+    project.countTarget,
+  ]);
+
+  const tpoNameBackgroundClass = useMemo(() => {
+    if (!project.allowDonations) return `${style.noDonation}`;
+    if (isTopProject && isApproved) return `${style.tpoBackground}`;
+    return '';
+  }, [isTopProject, isApproved, project.allowDonations]);
+
+  const progressBarClass = useMemo(() => {
+    return `${style[getProjectCategory(project)]}`;
+  }, [
+    project.purpose,
+    project.purpose === 'trees' && (project.isTopProject, project.isApproved),
+    project.allowDonations,
+  ]);
+
+  const handleClick = () => {
+    if (embed === 'true') {
+      window.open(`/t/${project.tpo.slug}`, '_top');
+    } else {
+      router.push(`/t/${project.tpo.slug}`);
+    }
+  };
+
+  const commonProps: CommonProps = {
+    slug: project.slug,
+    isApproved: isApproved,
+    isTopProject: isTopProject,
+    allowDonations: project.allowDonations,
+    purpose: project.purpose,
+  };
+  const imageProps: ImageSectionProps = {
+    ...commonProps,
+    projectName: project.name,
+    image: project.image,
+    ecosystem: ecosystem,
+    showPopup: showPopup,
+    projectReviews: project.reviews,
+    classification: (project as TreeProjectConcise).classification,
+  };
+  const projectInfoProps: ProjectInfoProps = {
+    ...commonProps,
+    unitType: project.unitType,
+    countPlanted: (project as TreeProjectConcise).countPlanted,
+    unitCost: project.unitCost,
+    country: project.country,
+    currency: project.currency,
+    ...(project.unitType === 'm2'
+      ? {
+          unitsContributed:
+            project.unitsContributed?.m2 || project.unitsContributed?.tree, //  planet-sdk need to be updated to include unitsContributed type
+        }
+      : null),
```

Comment:
This logic has an issue:

1. `project.unitType` does not predict what is within `project.unitsContributed`.
2. If the `unitType === 'tree'`, you are not passing on `unitsContributed`. Stop using `countPlanted`.